### PR TITLE
Add GCC problem matcher for GitHub Actions

### PR DESCRIPTION
**Pull request by cmarqu**

### CODE REVIEW

**Comment by ktbarrett** [comment]
Could we see a commit add an example warning added to the C++ to show that it works? Then you can add a revert commit for the warning code, just so a pointer to the failing action is recorded in the PR for posterity.

File: cocotb/share/lib/simulator/simulatormodule.cpp [hunk 1]
```diff
@@ -34,6 +34,7 @@
  * Uses GPI calls to interface to the simulator.
  */
 
+int unused;
```
Comment (ktbarrett) [inline cocotb/share/lib/simulator/simulatormodule.cpp:37]: Doesn't seem to work?

Comment (cmarqu) [reply cocotb/share/lib/simulator/simulatormodule.cpp:37]: Yeah, I need to find the time to work on this further.

**Comment by imphil** [comment]
Let's close this for the time being, we can always reopen if someone finds more time to work on this idea.

**Pull request closed without merging**